### PR TITLE
nawk: 20230911 -> 20240311

### PR DESCRIPTION
**Pull request by afh**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/na/nawk/package.nix [hunk 1]
```diff
@@ -8,13 +8,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "nawk";
-  version = "20230911";
+  version = "20240311";
 
   src = fetchFromGitHub {
     owner = "onetrueawk";
     repo = "awk";
     rev = "2ndEdition";
```
Comment (yayayayaka) [inline pkgs/by-name/na/nawk/package.nix:16]:
```suggestion
    rev = finalAttrs.version;
```

**Review by yayayayaka — CHANGES_REQUESTED** [review]
I am getting a hash mismatch when building the update:

```console
$ nixpkgs-review pr 295425
[...]
error: hash mismatch in fixed-output derivation '/nix/store/a77n67mz1mpri37q1yl0h64r65ydncbk-source.drv':
         specified: sha256-4iAQR8djhhp5Yn4H1IdzotQLY0d/Gz/vNQPfAUNQV0A=
            got:    sha256-9SxeYsTFrsJ+Cg43QjQihi2Ij0qdJvTnyGGhUHJHjuU=
error: 1 dependencies of derivation '/nix/store/afn25avpf3r87d925431cpij0xawvgc3-nawk-20240311.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2lg0a0327wslq4d8dv20f9v6qfs8ixj4-noweb-2.13.drv' failed to build
error: 2 dependencies of derivation '/nix/store/2xmchyivqkrmbfxqr3imfks6rpw84xls-review-shell.drv' failed to build
[...]
```

It seems, you forgot to update the `rev`.

File: pkgs/by-name/na/nawk/package.nix (hunk 1 above)
Comment (afh) [reply pkgs/by-name/na/nawk/package.nix]: Ohai @yayayayaka 👋😃 Thanks for catching this!

**Review by yayayayaka — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by afh** [comment]
Thanks for the review everyone and @yayayayaka also for merging, much appreciated 😃

**Pull request merged**